program/src/matching.rs

```diff
@@ -1610,3 +1782,46 @@ impl<'a> Book<'a> {
         Ok(())
     }
 }
+
+fn determine_ref_vars<'a>(
+    program_id: &Pubkey,
+    mango_group: &MangoGroup,
+    mango_group_pk: &Pubkey,
+    mango_cache: &MangoCache,
+    mango_account: &MangoAccount,
+    referrer_mango_account_ai: Option<&'a AccountInfo>,
+    now_ts: u64,
+) -> MangoResult<(I80F48, Option<RefMut<'a, MangoAccount>>)> {
+    let mngo_index = match mango_group.find_token_index(&mngo_token::id()) {
+        None => return Ok((ZERO_I80F48, None)),
+        Some(i) => i,
+    };
+
+    let mngo_cache = &mango_cache.root_bank_cache[mngo_index];
+    let mngo_deposits = mango_account.get_native_deposit(mngo_cache, mngo_index)?;
```

Comment:
is this guaranteed to produce 0 if the users doesn't have any MNGO and the cache is potentially out of date?

Reply:
Yes. If the user's MNGO deposit is non-zero then the rootbank cache will be checked already in `place_perp_order`. If  it's zero then cache may be out of date, but it doesn't matter because 0 * index = 0. 

I will add a comment here.